src/main/java/org/janelia/saalfeldlab/paintera/control/IdSelector.java

```diff
@@ -53,10 +67,92 @@ public MouseClickFX appendFragmentWithMaximumCount(final String name, final Pred
 		return new MouseClickFX(name, new AppendFragmentWithMaximumCount(), eventFilter);
 	}
 
-	public void toggleLock(
-			final SelectedIds selectedIds,
-			final FragmentSegmentAssignment assignment,
-			final LockedSegments lock)
+	// TODO: use unique labels to collect all ids; caching
+	public void selectAll()
+	{
+		final TLongSet allIds = new TLongHashSet();
+		if (source.getDataType() instanceof LabelMultisetType)
+			selectAllLabelMultisetType(allIds);
+		else
+			selectAllPrimitiveType(allIds);
+		LOG.info("Collected {} ids", allIds.size());
+		selectedIds.activate(allIds.toArray());
+	}
+
+	private void selectAllLabelMultisetType(final TLongSet allIds)
+	{
+		@SuppressWarnings("unchecked")
+		final RandomAccessibleInterval<LabelMultisetType> data = (RandomAccessibleInterval<LabelMultisetType>)
+				source.getDataSource(0, source.getNumMipmapLevels() - 1);
+
+		final Cursor<LabelMultisetType> cursor = Views.iterable(data).cursor();
+		while (cursor.hasNext())
+		{
+			final LabelMultisetType lmt = cursor.next();
+			for (final Entry<Label> entry : lmt.entrySet())
+			{
+				final long id = entry.getElement().id();
+				if (foregroundCheck.test(id))
+					allIds.add(id);
+			}
+		}
+	}
+
+	private void selectAllPrimitiveType(final TLongSet allIds)
+	{
+		LOG.warn("Label data is stored as primitive type, looping over full resolution data to collect all ids -- SLOW");
```

Comment:
Maybe we can make this cancellable (e.g. if the user hits escape/clicks an id etc.). This should not be part of this PR but maybe add a // TODO here